libvast/src/system/make_source.cpp

```diff
@@ -150,10 +142,19 @@ make_source(caf::actor_system& sys, const std::string& format,
   VAST_ASSERT(src);
   // Attempt to parse the remainder as an expression.
   if (!inv.arguments.empty()) {
-    auto expr = parse_expression(inv.arguments.begin(), inv.arguments.end());
-    if (!expr)
-      return expr.error();
-    send_to_source(src, atom::normalize_v, std::move(*expr));
+    auto query_result
+      = system::parse_query(inv.arguments.begin(), inv.arguments.end());
+    if (!query_result)
+      return query_result.error();
+    if (query_result->second) {
+      return caf::make_error(ec::invalid_configuration,
+                             fmt::format("{} parsed a pipeline string which "
+                                         "should "
+                                         "be impossible",
+                                         inv.full_name));
```

Comment:
Same deal as for the counter. This is where you will end up implementing support for import pipelines, though, which from this point on should be a really small change to make.

Reply:
Improved error message in https://github.com/tenzir/vast/pull/2877/commits/d0461071a5c0487d3d189be1978bd17cc56322e9